### PR TITLE
Merge/0.11.x to develop - #745

### PR DESCRIPTION
**Pull request by jboreiko**
The result of this merge is the inclusion of #745 into develop.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/768)

### CODE REVIEW

**Comment by gbrova** [comment]
LGTM.  

As I understand, the bulk of 0.11.x -> develop already happened in other PRs. This is just (1) adding the `CassandraKeyValueServiceConfigTest` test, (2) minor docs change, and (3) minor refactor to the `usingSsl` function.

**Comment by gbrova** [comment]
Disregard my last comment, was looking at the wrong thing. 

Yes, we need to merge this.

File: atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2015 Palantir Technologies
```
Comment (gbrova) [inline atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceConfigTest.java]: 2016

File: atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+public class CassandraKeyValueServiceConfigTest {
+    private static final InetSocketAddress SERVER_ADDRESS = InetSocketAddress.createUnresolved("localhost", 9160);
+    private static final SslConfiguration SSL_CONFIGURATION = SslConfiguration.of(Paths.get("./trustStore.jks"));
+
+    private static final ImmutableCassandraKeyValueServiceConfig CASSANDRA_CONFIG = ImmutableCassandraKeyValueServiceConfig.builder()
+            .addServers(SERVER_ADDRESS)
+            .replicationFactor(1)
+            .keyspace("atlasdb")
+            .build();
+
```
Comment (gbrova) [inline atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceConfigTest.java]: nit: Extra space

File: atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @Test
+    public void usingSslIfSslParamNotPresentAndSslConfigurationPresent() {
+        assertTrue(CASSANDRA_CONFIG.withSslConfiguration(SSL_CONFIGURATION).usingSsl());
+    }
+
+    @Test
+    public void notUsingSslIfSslParamNotPresentAndSslConfigurationNotPresent() {
+        assertFalse(CASSANDRA_CONFIG.usingSsl());
+    }
```
Comment (hsaraogi) [inline atlasdb-cassandra-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceConfigTest.java]: usingSslIfSslParamTrueAndSslConfigurationPresent??

**Pull request merged**